benchmark/benchmark.py

```diff
@@ -0,0 +1,229 @@
+# flake8: noqa: E402
+import argparse
+import bz2
+import csv
+import datetime
+import json
+import os
+import sys
+from io import StringIO
+from pathlib import Path
+
+from matplotlib import pyplot as plt  # type: ignore
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.dirname(SCRIPT_DIR))
+
+from eyecite import clean_text, get_citations
+
+csv.field_size_limit(sys.maxsize)
+
+root = Path(__file__).parent.absolute()
+
+MAX_ROWS_IN_MD = 51
+
+
+class Benchmark(object):
+    """Benchmark the different eyecite branches"""
+
+    def get_filepath(self, filename):
+        return Path.joinpath(root, filename)
+
+    def generate_branch_report(self, branch: str) -> None:
+        """Generate Report
+
+        :param branch: Is a branch from main or not
+        :return: None
+        """
+        zipfile = bz2.BZ2File(self.get_filepath("bulk-file.csv.bz2"))
+        csv_data = csv.DictReader(
+            StringIO(zipfile.read().decode()), delimiter=","
+        )
```

Comment:
I'm not super familiar with the bz2 library, but I suspect this leaves an open file handle, which, admittedly, doesn't matter since we're running this on some docker image in CI, but still, might I suggest the somewhat tidier:

```suggestion
        with bz2.BZ2File(self.get_filepath("bulk-file.csv.bz2")) as zipfile:
            csv_data = csv.DictReader(zipfile.read().decode(), delimiter=",")
```

I think this should be a bit more performant too, but I'm just guessing.

Reply:
this was for legibility